hawkbit-repository/hawkbit-repository-jpa/src/main/java/org/eclipse/hawkbit/repository/jpa/aspects/ExceptionMappingAspectHandler.java

```diff
@@ -84,88 +74,50 @@
 
-            if (mappedEx.isAssignableFrom(translatedAccessException.getClass())) {
+            if (mappedEx.isAssignableFrom(ex.getClass())) {
                 if (!EXCEPTION_MAPPING.containsKey(mappedEx.getName())) {
                     LOG.error("there is no mapping configured for exception class {}", mappedEx.getName());
                     mappingException = new GenericSpServerException(ex);
                 } else {
                     mappingException = (Exception) Class.forName(EXCEPTION_MAPPING.get(mappedEx.getName()))
                             .getConstructor(Throwable.class).newInstance(ex);
                 }
+                LOG.trace("mapped exception {} to {}", ex.getClass(), mappingException.getClass());
                 break;
             }
         }
 
-        LOG.trace("mapped exception {} to {}", translatedAccessException.getClass(), mappingException.getClass());
         throw mappingException;
     }
 
-    private DataAccessException translateEclipseLinkExceptionIfPossible(final Exception exception) {
-        final DataAccessException translatedAccessException = jpaVendorAdapter.getJpaDialect()
-                .translateExceptionIfPossible((RuntimeException) exception);
-        return translateSQLStateExceptionIfPossible(translatedAccessException);
-
-    }
-
-    /**
-     * There is no EclipseLinkExceptionTranslator. So we have to check and
-     * translate the exception by the sql error code. Luckily, there we can use
-     * {@link SQLStateSQLExceptionTranslator} if we can get a
-     * {@link SQLException}.
-     *
-     * @param accessException
-     *            the base access exception from jpa
-     * @return the translated accessException
-     */
-    private DataAccessException translateSQLStateExceptionIfPossible(final DataAccessException accessException) {
-        if (!(accessException instanceof JpaSystemException)) {
-            return accessException;
-        }
-
-        final SQLException ex = findSqlException((JpaSystemException) accessException);
-
-        if (ex == null) {
-            return accessException;
-        }
-
-        return sqlStateExceptionTranslator.translate(null, null, ex);
-    }
-
-    private static SQLException findSqlException(final JpaSystemException jpaSystemException) {
-        Throwable exception = jpaSystemException.getCause();
-        while (exception != null) {
+    private static Exception findConstraintViolationException(final Exception rex) {
+        Throwable exception = rex;
+        do {
             final Throwable cause = exception.getCause();
-            if (cause instanceof SQLException) {
-                return (SQLException) cause;
+            if (cause instanceof javax.validation.ConstraintViolationException) {
+                return (Exception) cause;
             }
             exception = cause;
-        }
-        return null;
+        } while (exception != null);
+
+        return rex;
```

Comment:
Is this intended not return a null value? So the `TransactionSystemException` is always thrown and will not translated.